src/resolver.js

```diff
@@ -4,14 +4,17 @@ import {normalizeSwagger} from './helpers'
 
 export function makeFetchJSON(http) {
   return (docPath) => {
-    return http({
+    return Promise.resolve(http({
       url: docPath,
       loadSpec: true,
       headers: {
         Accept: 'application/json'
       }
+    }))
+    .then((res) => {
+      // To allow overriding with spies
```

Comment:
The spy in this case is the `createSpy()` exposed in `expect`
Its to allow putting in a mock'd version of `http` ( as referred to in previous comments ).

This is here, so that we don't need to provide a `.body` property on what the mock'd  `http` method returns. Given that this will be rarely used. I don't mind removing it. It was a workaround to start with... now that I know we only test in nodejs... we can continue to use `xmock`